submissions/Ant-C-tech/friends-app/js/app.js

```diff
@@ -0,0 +1,288 @@
+        //Sort
+    switch (sortParameter) {
+        case '0-100':
+            CURRENT_FRIENDS.sort(byField('dob', 'age'))
+            break
+        case '100-0':
+            CURRENT_FRIENDS.sort(byField('dob', 'age')).reverse()
+            break
+        case 'name_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'first'))
+            break
+        case 'name_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'first')).reverse()
+            break
+        case 'lastName_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'last'))
+            break
+        case 'lastName_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'last')).reverse()
+            break
+
+        default:
+            break
+    }
+
+    changeContent(createFriendsScreen(CURRENT_FRIENDS), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function resetFilter() {
+    //In case user enter to app through sidenav (not using startBtn)
+    if (!isMusicStoppedByUser) {
+        playMusic()
+    }
+    showSearchBar()
+    closeSideNav()
+
+    CURRENT_FRIENDS = []
+    if (document.querySelector("input[type='radio'][name='gender']:checked")) {
+        document.querySelector("input[type='radio'][name='gender']:checked").checked = false
+    }
+    if (document.querySelector("input[type='radio'][name='sort']:checked")) {
+        document.querySelector("input[type='radio'][name='sort']:checked").checked = false
+    }
+    changeContent(createFriendsScreen(FRIENDS_SOURCE), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function closeSideNav() {
+    SEARCH_INPUT.value = ''
+    document.querySelector('.search__hint').classList.remove('active')
+}
+
+function activateSideNav() {
+    const sideNavIco = document.querySelectorAll('.sidenav')
+    M.Sidenav.init(sideNavIco, {})
+}
+
+function playMusic() {
+    APP_AUDIO.play()
+    MUSIC_BTN.classList.add('musicBtn-active')
+    MUSIC_BTN.classList.add('musicBtn-animation')
+    MUSIC_BTN.addEventListener('click', stopMusic, { once: true })
+}
+
+function stopMusic() {
+    APP_AUDIO.pause()
+    MUSIC_BTN.classList.remove('musicBtn-animation')
+    MUSIC_BTN.addEventListener('click', playMusic, { once: true })
+    isMusicStoppedByUser = true
+}
+
+function getRandomIntInclusive(min, max) {
+    min = Math.ceil(min)
+    max = Math.floor(max)
+    return Math.floor(Math.random() * (max - min + 1) + min)
+}
+
+function byField(fieldName, subFieldName) {
+    return (a, b) => a[fieldName][subFieldName] > b[fieldName][subFieldName] ? 1 : a[fieldName][subFieldName] < b[fieldName][subFieldName] ? -1 : 0;
+}
+
+//noUiSlider
+function createRangeSlider() {
+    const rangeSlider = document.getElementById('test-slider')
+    const rangeSliderSettings = {
+        start: [20, 80],
+        connect: true,
+        step: 1,
+        orientation: 'horizontal',
+        range: {
+            'min': 0,
+            'max': 100
+        },
+        format: wNumb({
```

Comment:
What is `wNumb`? 🤔

Reply:
Is it a good idea to leave the link on API in comments? Maybe it would be convinient in this case?